scipy/io/matlab/mio5_utils.pyx

```diff
@@ -474,11 +474,17 @@ cdef class VarReader5:
         Specializes ``read_element``
         '''
         cdef:
-            cnp.uint32_t mdtype, byte_count
-            void *ptr
+            cnp.uint32_t mdtype, byte_count, i
+            void* ptr
+            unsigned char* byte_ptr
             object data
         data = self.read_element(&mdtype, &byte_count, &ptr)
-        if mdtype != miINT8:
+        if mdtype == miUTF8:  # Some badly-formed .mat files have utf8 here
+            byte_ptr = <unsigned char*> ptr
+            for i in range(byte_count):
+                if byte_ptr[i] > 127:
+                    raise ValueError('Non ascii int8 string ' + str(data))
```

Comment:
``str(data)`` doesn't look right here - this will result in:

    UnicodeEncodeError: 'ascii' codec can't encode characters in position xxx-yyy: ordinal not in range(128)